### ISSUE TITLE
move to a public scoped package

### ISSUE THREAD

**Issue by dominicbarnes** [segmentio/eslint-config#1]
Maybe this could be our first usage of a scoped package?

`@segment/eslint-config`

I'm not sure all the implications of making such a transition, but I know we need to start on it eventually.

**Comment by calvinfo** [comment]
Sounds like a good idea to me! @vinceprignano has been doing a bunch of work to migrate us over to the private npm everywhere so soon our internal stuff should be scoped as well.

**Comment by ndhoule** [comment]
Sounds good to me! ESLint added support for this very recently:

https://github.com/eslint/eslint/commit/f56ff48a165119557938a2cd641c6f6dd829003c

Looks like we'd need to make sure all packages used `eslint@^v0.21.2`—maybe make it a `peerDependency`.

**Comment by ndhoule** [comment]
Steps to migrate here:
- [ ] Update config
- [ ] Publish to private npm
- [ ] `npm deprecate` old package
- [ ] Update all packages that depend on this to depend on scoped package

**Comment by ndhoule** [comment]
realized the scoped name ends up somewhat awkward:

`@segment/segment/browser`

now I'm wondering if it's even worth doing

reference: https://github.com/eslint/eslint/issues/2594

**Comment by dominicbarnes** [comment]
In my own personal projects, I've actually opted to do several config repos, each geared toward a specific context. ([notes](https://github.com/dominicbarnes/eslint-config/blob/master/README.md))

The main downside to the approach I have is that I've opted to use a lot of `peerDependencies`, so you typically install the entire chain in a given app. It's a pain, but only a 1-time pain. (I'm not sure how well using `dependencies` works, since the nesting could complicate resolution by eslint)

Despite the shortcomings, I _really_ like the finished product, where I can easily maintain config for various contexts, without unnecessary `peerDependencies` that need to be installed. (such as repos that don't use babel/deku being forced to install `eslint-plugin-react` to satisfy the `peerDependency`)
